### ISSUE TITLE
Arduboy3D/Catacombs of the Damned no longer boots

### ISSUE THREAD

**Issue by thepulserifle** [MiSTer-devel/Arduboy_MiSTer#4]
In previous releases, Arduboy3D has been able to be played to completion (with the occasional reset). However, in the 3/23 build, it no longer launches.

**Comment by uXeBoy** [comment]
Yes, there were some changes made to the AVR core which resulted in a [lot more games](https://github.com/uXeBoy/ArduboyCollection/commit/fc9872ac67c99a3386243bcfd60fb8037ecff538) being playable that didn't work previously... but an unfortunate side-effect was that a few games, like Arduboy3D, stopped working!

You can use the Arduboy3D.bin file from here though and it will work fine:

https://github.com/uXeBoy/ArduboyCollection/blob/master/Arduboy3D.bin?raw=true

This is a re-compiled version of the game, the only difference being that I used a slightly [different library](https://github.com/uXeBoy/ArduboyNewTones) implementation for the sound. Why this makes a difference is still up for investigation though... I know MorgothCreator was working on a debug module to try and nail down these issues by being able to step-through instructions until the problem ones can be found - I am more than happy for any interested FPGA gurus to step in and assist with the bug-hunt though!

**Comment by cmstar0** [comment]
@uXeBoy @sorgelig 

This is fixed by #17 — a CPU bug where an STS instruction immediately following an IN/OUT to SREG (the standard cli()-protected atomic-read pattern from avr-libc) could silently clobber SREG, including the global interrupt-enable bit. Catacombs of the Damned hits this during its ROM/dungeon-generation boot sequence, which is why it hung before ever reaching gameplay.

Verified with an isolated hardware test — #17 alone, no other changes — boots past the progress bar and is playable. Should be safe to close (I don't have privileges to do so.)
